Katib v1beta1 documentation update

andreyvelich opened the pull request:
Related: https://github.com/kubeflow/katib/issues/1360.
Blocked by: https://github.com/kubeflow/manifests/pull/1593.
I updated docs for Katib v1beta1. I made few changes:

1. Update reference docs for v1beta1. @8bitmp3 @RFMVasconcelos Do we want to have v1alpha3 reference since all docs are written for v1beta1 ?
1. Fix few broken links.
1. Rename `hyperparameter-tuning` section to `katib` to make it clearer, since we are building AutoML project. What do you think about it @8bitmp3 @RFMVasconcelos ?
1. Add missing links for the examples, controller components, etc.
1. Update images.

I am still waiting for the pending PRs (I need to update docker images and few UI images., but it would be great if you can start to review it!

/assign @gaocegege @johnugeorge 
/cc @8bitmp3 @RFMVasconcelos

RFMVasconcelos commented:
Hi @andreyvelich ! 

Thank you for this update! Replying by point:

> Update reference docs for v1beta1. @8bitmp3 @RFMVasconcelos Do we want to have v1alpha3 reference since all docs are written for v1beta1 ?

You're asking if we want to keep a v1alpha3 version of the docs, now that we're updating Katib to v1beta1, correct? 
If so:
- When will these docs be effective? If we're launching v1beta1 with Kubeflow v1.2, would it make sense to include this in v1.2 of the docs?
- If so, we can simply add a reference to v1alpha3 saying that docs can be found in Kubeflow v1.1 docs (with a link)

WDYT?

> Rename hyperparameter-tuning section to katib to make it clearer, since we are building AutoML project. What do you think about it @8bitmp3 @RFMVasconcelos ?

LGTM.

RFMVasconcelos reviewed (commented):
One other point was the name change from Katib to something else. If we plan to do that, should we already make changes accordingly? 

Any decision on that front?

andreyvelich commented:
Thank you for the comment @RFMVasconcelos!

> You're asking if we want to keep a v1alpha3 version of the docs, now that we're updating Katib to v1beta1, correct?
> If so:
> 
> * When will these docs be effective? If we're launching v1beta1 with Kubeflow v1.2, would it make sense to include this in v1.2 of the docs?

These docs will be effective after we merge: https://github.com/kubeflow/manifests/pull/1593.
Including docs in V1.2 makes sense to me. I assume that website `master` branch will be point to 1.2 Kubeflow release, right ? If so, I am writing these docs to the appropriate branch.

> * If so, we can simply add a reference to v1alpha3 saying that docs can be found in Kubeflow v1.1 docs (with a link)

I think, it's a good idea. I will add v1alpha3 reference docs and add link to 1.1. Is there any way to get link for 1.1 docs? I found only 1.0 link: https://v1-0-branch.kubeflow.org/.

andreyvelich commented:
> One other point was the name change from Katib to something else. If we plan to do that, should we already make changes accordingly?
> 
> Any decision on that front?

I think we can stick with Katib name for Kubeflow 1.2. 
If we decide to rename the project after 1.2, I will submit separate PR to make changes in the docs.

WDYT @gaocegege @johnugeorge ?

andreyvelich commented:
Preview: https://deploy-preview-2312--competent-brattain-de2d6d.netlify.app/.

I have updated doc with v1alpha3 reference.

/cc @RFMVasconcelos @gaocegege @johnugeorge

RFMVasconcelos reviewed (commented):
Looking great! 

You are indeed writing to the right branch! What we need to do is create the v1.1 branch and update master to be v1.2. 

Let me create an issue about it :)

RFMVasconcelos commented:
Oh, it looks like there is one already -> #2322

andreyvelich commented:
Thanks @RFMVasconcelos.
Do you have any comments/objections on the doc changes in this PR?

andreyvelich commented:
@RFMVasconcelos @8bitmp3 @gaocegege @johnugeorge I removed WIP status, since https://github.com/kubeflow/manifests/pull/1593 is ready to merge. If you have any objections and comments on this PR, please let me know.
If not, we can merge it and start work on other Katib documentation PRs

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+                    
++++
+
+This page describes information about
+[Katib config](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-config.yaml).
+
+Katib config is the Kubernetes
+[Config Map](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) that contains information about:
```
Thank you for the PR @andreyvelich 

Hi, what do you think of this suggestion? 
- "refactor" to 'This page describes Katib config — the Kubernetes Config Map that contains information about:'

```suggestion
This page describes
[Katib config](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-config.yaml) — 
the Kubernetes
[Config Map](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) that contains information about:
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+You can edit this Config Map even after deploying Katib.
+
+If you deploy Katib in Kubeflow namespace, to edit Katib config run this:
```
Maybe just change the order here, use a different (present continuous) tense, and "the" (the definite article) (I checked with the K8s [guide](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/), as I'm not 100% sure about these):

```suggestion
If you are deploying Katib in the Kubeflow namespace, run this command to edit your Katib config:
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+You can edit this Config Map even after deploying Katib.
+
+If you deploy Katib in Kubeflow namespace, to edit Katib config run this:
+
+`kubectl edit configMap katib-config -n kubeflow`
+
```
Suggestion - use code style/code font/code block/codefence (K8s style [guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-inline-code-commands-and-api-objects), Google style [guide](https://developers.google.com/style/code-in-text#formatting-a-command):

```suggestion

    ```shell
    kubectl edit configMap katib-config -n kubeflow
    ```
```
(without the tabs that indent the code block - they render differently if I remove them)

andreyvelich replied:
@8bitmp3 Should we use ```shell everywhere?
e.g. https://github.com/kubeflow/website/blame/master/content/en/docs/components/hyperparameter-tuning/hyperparameter.md#L144 ?

andreyvelich replied:
@8bitmp3 I've just seen you comment: https://github.com/kubeflow/website/pull/2320/files#r516300401 about it.
Will do the change in this PR.

andreyvelich replied:
Done in https://github.com/kubeflow/website/pull/2312/commits/fe91e741a3f97bdd81e9e34b63a409124f9f933c.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+These settings are related to Katib metrics collectors, where:
+
+- key = `metrics-collector-sidecar`
+- value = corresponding JSON settings for each metrics collector kind.
```
Formatting suggestion:

```suggestion
- key: `metrics-collector-sidecar`
- value: corresponding JSON settings for each metrics collector kind
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+All of these settings except **`image`** can be omitted. If you don't specify any other settings, default value is set.
+
+1. `image` - Docker image for the `File` metrics collector's container.
```
```suggestion
1. `image` - a Docker image for the `File` metrics collector's container.
```

andreyvelich replied:
Agree, changed.
@8bitmp3 I made all list items with not capital letter to be consistent (e.g. https://github.com/kubeflow/website/blame/62aba5ce7a065eb2b60d6b7884f4ab075be33fcc/content/en/docs/components/katib/katib-config.md#L66)
WDYT ?

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
````diff
@@ -0,0 +1,211 @@
+}
+```
+
+All of these settings except **`image`** can be omitted. If you don't specify any other settings, default value is set.
````
```suggestion
All of these settings except **`image`** can be omitted. If you don't specify any other settings, a default value is set automatically.
```

(Does this proposal make sense?)

andreyvelich replied:
Agree, changed. As well, I added the same sentence for the https://github.com/kubeflow/website/pull/2312#discussion_r516258436, I'd like your first suggestion.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+1. `imagePullPolicy` - [Image pull policy](https://kubernetes.io/docs/concepts/configuration/overview/#container-images) for the `File` metrics collector's container.
+
+   Default value is `IfNotPresent`.
```
```suggestion
   The default value is `IfNotPresent`.
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+1. `image` - Docker image for the `File` metrics collector's container.
+
+   **Must be specified**.
```
```suggestion
   **The following values must be specified**:
```

(Does this suggestion make sense?)

andreyvelich replied:
That is related to `image`.
@8bitmp3 What do you think about this:
1. `image` - a Docker image for the `File` metrics collector's container (**must be specified**).

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+   for the `File` metrics collector's container. In the above example you can see how to specify
+   `limits` and `requests`. Currently, you can specify only `memory`, `cpu` and `ephemeral-storage` resource.
+
+   Default values for the `requests` are:
```
```suggestion
   The default values for the `requests` are:
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+   - `cpu = 50m`.
+   - `ephemeral-storage = 500Mi`.
+
+   Default values for the `limits` are:
```
```suggestion
   The default values for the `limits` are:
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+These settings are related to Katib suggestions, where:
+
+- key = `suggestion`
+- value = corresponding JSON settings for each algorithm name.
```
Suggestion:

```suggestion
- key: `suggestion`
- value: corresponding JSON settings for each algorithm name
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+- key = `suggestion`
+- value = corresponding JSON settings for each algorithm name.
+
+If you want to use new algorithm, you must update Katib config with the new suggestion.
+
+Example for the `random` algorithm with all settings:
```
Suggestion:

```suggestion
If you want to use a new algorithm, you need to update the Katib config. For example, using a `random` algorithm with all settings looks as follows:
```

8bitmp3 replied:
There's a Markdown that described the `random` algorithm - maybe we should it link it at the first mention in this doc? Up to you.

andreyvelich replied:
If you want to use a new algorithm, you need to update the Katib config.

andreyvelich replied:
> There's a Markdown that described the `random` algorithm - maybe we should it link it at the first mention in this doc? Up to you.

@8bitmp3 Do you mean this doc: https://deploy-preview-2312--competent-brattain-de2d6d.netlify.app/docs/components/katib/experiment/#random-search ?

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
````diff
@@ -0,0 +1,211 @@
+```
+
+All of these settings except **`image`** can be omitted.
+If you don't specify any other settings, default value is set.
````
```suggestion
If you don't specify any other settings, a default value is set to each setting.
```

(Does this suggestion make sense?)

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+All of these settings except **`image`** can be omitted.
+If you don't specify any other settings, default value is set.
+
+1. `image` - Docker image for the suggestion's container with `random` algorithm.
```
```suggestion
1. `image` - a Docker image for the suggestion's container with a `random` algorithm.
```

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+
+1. `image` - Docker image for the suggestion's container with `random` algorithm.
+
+   **Must be specified**.
```
```suggestion
   **The following values must be specified**:
```

(Does this suggestion make sense?)

andreyvelich replied:
See comment: https://github.com/kubeflow/website/pull/2312#discussion_r516328626.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+           <td><code>suggestion-hyperopt</code></td>
+           <td><code>random</code>, <code>tpe</code></td>
+           <td><a href="https://github.com/hyperopt/hyperopt">Hyperopt</a> optimization framework</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-chocolate</code></td>
+           <td><code>grid</code>, <code>random</code>, <code>quasirandom</code>, <code>bayesianoptimization</code>, <code>mocmaes</code></td>
+           <td><a href="https://github.com/AIworx-Labs/chocolate">Chocolate</a> optimization framework</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-skopt</code></td>
+           <td><code>bayesianoptimization</code></td>
+           <td><a href="https://github.com/scikit-optimize/scikit-optimize">Scikit-optimize</a> optimization framework</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-goptuna</code></td>
+           <td><code>cmaes</code>, <code>random</code>, <code>tpe</code></td>
+           <td><a href="https://github.com/c-bata/goptuna">Goptuna</a> optimization framework</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-hyperband</code></td>
+           <td><code>hyperband</code></td>
+           <td><a href="https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1beta1/hyperband">Katib
+             Hyperband</a> implementation</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-enas</code></td>
+           <td><code>enas</code></td>
+           <td><a href="https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1beta1/nas/enas">Katib
+             ENAS</a> implementation</td>
+         </tr>
+         <tr>
+           <td><code>suggestion-darts</code></td>
+           <td><code>darts</code></td>
+           <td><a href="https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1beta1/nas/darts">Katib
+             DARTS</a> implementation</td>
+         </tr>
+       </tbody>
+     </table>
+   </div>
```
Should we wrap this in a code block with an `html` "tag"

```
    ```html
    ...
    ```
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+1. `imagePullPolicy` - [Image pull policy](https://kubernetes.io/docs/concepts/configuration/overview/#container-images)
+   for the suggestion's container with `random` algorithm.
+
+   Default value is `IfNotPresent`.
```
```suggestion
   The default value is `IfNotPresent`.
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+   for the suggestion's container with `random` algorithm. In the above example you can see how to specify
+   `limits` and `requests`. Currently, you can specify only `memory`, `cpu` and `ephemeral-storage` resource.
+
+   Default values for the `requests` are:
```
```suggestion
   The default values for the `requests` are:
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+   - `cpu = 50m`.
+   - `ephemeral-storage = 500Mi`.
+
+   Default values for the `limits` are:
```
```suggestion
   The default values for the `limits` are:
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+   In the above example, `random-sa` service account is attached for each Experiment's
+   suggestion with `random` algorithm until you change or delete this service account from the Katib config.
+
+   By default suggestion pod doesn't have any specific service account.
+   In that case, suggestion's pod uses [default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server) service account.
```
Maybe we can merge these:


```suggestion
   By default, the suggestion pod doesn't have any specific service account,
   in which case, the pod uses the 
   [default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server) service account.
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -87,83 +86,82 @@ part of the form for submitting a NAS job from the Katib UI:
 
-  * **Kfctl** is the Kubeflow CLI that you can use to install and configure 
-    Kubeflow. Read about kfctl in the guide to 
+  - **Kfctl** is the Kubeflow CLI that you can use to install and configure
```
```suggestion
  - **kfctl** is the Kubeflow CLI that you can use to install and configure
```

(do we use it with a capitalized letter K? Not 100% sure)

andreyvelich replied:
Agree, changed.

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,211 @@
+1. Current [metrics collectors](/docs/components/katib/experiment/#metrics-collector) (`key = metrics-collector-sidecar`)
+1. Current [algorithms](/docs/components/katib/experiment/#search-algorithms-in-detail) (suggestions) (`key = suggestion`).
+
+Katib Config Map must be deployed in
+[`KATIB_CORE_NAMESPACE`](/docs/components/katib/env-variables/#katib-controller)
+namespace with `katib-config` name. Katib controller parses Katib config when you submit Experiment.
```
Suggestions:

```suggestion
The Katib Config Map must be deployed in the
[`KATIB_CORE_NAMESPACE`](/docs/components/katib/env-variables/#katib-controller)
namespace with the `katib-config` name. The Katib controller parses the Katib config when you submit your experiment.
```

(or maybe there's a reason it was written as `Experiment`)

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -5,79 +5,78 @@ weight = 10
 
-Use Katib for automated tuning of your machine learning (ML) model's 
-hyperparameters and architecture.
+Katib is Kubernetes-native project for automated machine learning (AutoML).
+Use Katib for automated tuning of your machine learning (ML) model's hyperparameters and architecture.
```
Suggestion:

```suggestion
Katib is a Kubernetes-native project for automated machine learning (AutoML).
You can use Katib for automated tuning of your machine learning (ML) model's hyperparameters and architecture.
```

8bitmp3 reviewed (changes requested):
@andreyvelich Thanks for the PR!

I've added a few suggestions and comments. Feel free to disregard them 👍 

Maybe in the Katib explanation in `(/components/katib/overview.md` we could also add a small link for all the new ML users about AutoML (e.g. https://www.fast.ai/2018/07/16/auto-ml2/ and/or Microsoft Azure, Google Cloud and AWS AutoML solution docs? 

Thank you for your help!

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -5,79 +5,78 @@ weight = 10
 
 For example, the following graph from Katib shows the level of validation accuracy
-for various combinations of hyperparameter values (learning rate, number of 
+for various combinations of hyperparameter values (learning rate, number of
 layers, and optimizer):
```
```suggestion
for various combinations of hyperparameter values (the learning rate, the number of
layers, and the optimizer):
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -5,79 +5,78 @@ weight = 10
-*Hyperparameters* are the variables that control the model training process. 
-For example: 
+_Hyperparameters_ are the variables that control the model training process.
+For example:
```
```suggestion
They include:
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -5,79 +5,78 @@ weight = 10
-* Learning rate.
-* Number of layers in a neural network.
-* Number of nodes in each layer.
+- Learning rate.
+- Number of layers in a neural network.
+- Number of nodes in each layer.
```
```suggestion
- The learning rate.
- The number of layers in a neural network.
- The number of nodes in each layer.
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -87,83 +86,82 @@ part of the form for submitting a NAS job from the Katib UI:
-Katib is a Kubernetes-based system for hyperparameter tuning and neural 
-architecture search. Katib supports a number of ML frameworks, including 
+Katib is a Kubernetes-based system for hyperparameter tuning and neural
+architecture search. Katib supports a number of ML frameworks, including
 TensorFlow, MXNet, PyTorch, XGBoost, and others.
 
-The [Katib project](https://github.com/kubeflow/katib) is open source. 
+The [Katib project](https://github.com/kubeflow/katib) is open source.
 The [developer guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md)
 is a good starting point for developers who want to contribute to the project.
```
I think lines 87-95 about "the Katib project" should be right in the beginning of `overview.md` (see earlier comment).

andreyvelich commented:
@8bitmp3 Thank you so much for your review and suggestions!

>> Maybe in the Katib explanation in /components/katib/overview.md we could also add a small link for all the new ML users about AutoML (e.g. https://www.fast.ai/2018/07/16/auto-ml2/) and/or Microsoft Azure, Google Cloud and AWS AutoML solution docs?
Thank you for your help!
Update: also maybe we should move the "about" section up in overview.md such that:
Katib is a Kubernetes-native project for automated machine learning (AutoML) —
it's a system for hyperparameter tuning and neural architecture search. 
Katib supports a number of machine learning frameworks, including
TensorFlow, MXNet, PyTorch, XGBoost, and others.
The [Katib project](https://github.com/kubeflow/katib) is open source.
The [developer guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md)
is a good starting point for developers who want to contribute to the project.

It's great idea to move this section up.
What do you think about this introduction?

```

Katib is a Kubernetes-native project for automated machine learning (AutoML).
For example, to run hyperparameter tuning or neural architecture search.
Learn more about AutoML at [fast.ai](https://www.fast.ai/2018/07/16/auto-ml2/),
[Google Cloud](https://cloud.google.com/automl),
[Microsoft Azure](https://docs.microsoft.com/en-us/azure/machine-learning/concept-automated-ml#automl-in-azure-machine-learning),
[Amazon SageMaker](https://aws.amazon.com/blogs/aws/amazon-sagemaker-autopilot-fully-managed-automatic-machine-learning/)

Katib is the framework agnostic to machine learning frameworks.
It can tune hyperparameters of applications written in any language
of the users' choice and natively supports many machine learning frameworks,
such as TensorFlow, MXNet, PyTorch, XGBoost, and others.

The [Katib project](https://github.com/kubeflow/katib) is open source.
The [developer guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md)
is a good starting point for developers who want to contribute to the project.

This page introduces the concepts of hyperparameter tuning, neural
architecture search, and the Katib system as a component of Kubeflow.
```

andreyvelich commented:
@8bitmp3 Does ` ```html ` "tag" work for your HTML tables ?
For me, it doesn't render the table properly.
As well, I haven't found examples where we use this tag.
We just write `<div>` directly to Markdown files: https://github.com/kubeflow/website/blame/master/content/en/docs/about/style-guide.md#L60

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -383,7 +384,7 @@ If you installed Katib as part of Kubeflow, you can’t run experiments in Kubef
 training job operator, PyTorchJob:
 
-1. Download the pytorchjob-example.yaml file
+1. Download the pytorchjob-example.yaml file:
```
```suggestion
1. Download `pytorchjob-example.yaml`:
```

"...or the `...` file:" (as before)
("code font")

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -412,7 +413,7 @@ Follow the steps as described for the _random algorithm example_
 
-To delete Katib from Kubernetes cluster run::
+To delete Katib from Kubernetes cluster run:
 
```
@andreyvelich Hi!

On Lines 424-430 - please double check your hyperlinks. Also, can we refactor this for accessibility/inclusivity ("see" -> "refer to" or "go to" or similar)?

For example:

```
- Learn to configure and run your experiment with Katib in the
  [running an experiment](/docs/components/hyperparameter-tuning/experiment/) guide.
- About the [Katib configuration file (Katib config)](/docs/components/hyperparameter-tuning/katib-config/).
- How to [set up environment variables for each Katib component](/docs/components/hyperparameter-tuning/env-variables/).
```

@andreyvelich Please double check your links. I think `/docs/components/hyperparameter-tuning/...` work but the older `/docs/components/katib/`? don't (not any more, at least). Thanks!

8bitmp3 replied:
Update: The links are changing to `docs/components/kativ` - Got it!

8bitmp3 reviewed (changes requested):
@andreyvelich Hi!

On Lines 424-430 - can we refactor this for accessibility/inclusivity ("see" -> "refer to" or "go to" or similar)?

For example:

```
- Learn to configure and run your experiment with Katib in the
  [running an experiment](/docs/components/katib/experiment/) guide.
- About the [Katib configuration file (Katib config)](/docs/components/katib/katib-config/).
- How to [set up environment variables for each Katib component](/docs/components/katib/env-variables/).
```

Thanks!

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -5,14 +5,16 @@ weight = 50
 +++
 
-This page describes information about environment variables for each Katib component. If you want to change Katib installation, you can modify some of these variables.
+This page describes information about environment variables for each Katib component.
```
```suggestion
This guide is about environment variables for each Katib component.
```

andreyvelich replied:
SGTM, changed

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -5,14 +5,16 @@ weight = 50
 
-This page describes information about environment variables for each Katib component. If you want to change Katib installation, you can modify some of these variables.
+This page describes information about environment variables for each Katib component.
+If you want to change Katib installation, you can modify some of these variables.
```
```suggestion
If you want to change your Katib installation, you can modify some of these variables.
```

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -5,14 +5,16 @@ weight = 50
 
-In the tables below you can see description, default values and mandatory property for all environment variables in each Katib component.
-If variable has mandatory property, appropriate Katib component's manifest must have this environment variable.
+In the tables below you can see description, default values and mandatory property for
+all environment variables in each Katib component. If variable has mandatory property,
+appropriate Katib component's manifest must have this environment variable.
```
Suggestions:

- Change to plural (because of "all variables" per component)
- Add indefinite articles ("a")

```suggestion
In the tables below there are descriptions, default values and mandatory properties for
all environment variables in each Katib component. If a variable has a mandatory property,
you need to set the relevant environment variable in an appropriate Katib component's manifest.
```

(Please check if the meaning is still the same - we're giving instructions to "you" (the reader), so changing to active voice)

(It may also make sense to link this to 80 char per line)

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -5,14 +5,16 @@ weight = 50
 ## Katib Controller
 
-This is the table of environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/katib-controller/katib-controller.yaml) deployment.
+This is the table of environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-controller.yaml) deployment.
```
```suggestion
These are environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-controller.yaml) deployment:
```

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -71,7 +74,7 @@ If you want to use your own DB Manager to report Katib metrics, you can change `
 ## Katib UI
 
-This is the table of environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/ui/deployment.yaml) deployment.
+This is the table of environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml) deployment.
```
```suggestion
These are environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml) deployment:
```

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -171,21 +174,25 @@ This is the table of environment variables for the [Katib DB Manager](https://gi
 
-For the [Katib MySQL](https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/mysql-db/deployment.yaml) we set these environment variables:
+For the [Katib MySQL](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/mysql-db/deployment.yaml)
+we set these environment variables:
```
```suggestion
you need to set these environment variables:
```

Using the second person ("you" - the user) is preferred

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -171,21 +174,25 @@ This is the table of environment variables for the [Katib DB Manager](https://gi
 - `MYSQL_DATABASE` as `katib`.
 
-Check [here](https://github.com/docker-library/docs/tree/master/mysql#environment-variables) about all environment variables for the MySQL Docker image.
+Check [here](https://github.com/docker-library/docs/tree/master/mysql#environment-variables)
+about all environment variables for the MySQL Docker image.
```
```suggestion
You can refer to the list of 
[all environment variables](https://github.com/docker-library/docs/tree/master/mysql#environment-variables)
for the MySQL Docker image.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -6,32 +6,33 @@ weight = 30
 
-See the [YAML file for the random algorithm 
-example](https://github.com/kubeflow/katib/blob/master/examples/v1alpha3/random-example.yaml).
+See the [YAML file for the random algorithm
```
```suggestion
As a reference, you can use the YAML file of the [random algorithm
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
-  also known as the *feasible set* or the *solution space*.
-  In this section of the spec, you define the name and the distribution 
+- **parameters**: The range of the hyperparameters or other parameters that you
+  want to tune for your ML model. The parameters define the _search space_,
```
Please double check that "ML" has been predefined in this doc (it's a small thing and we know it's "machine learning" (what else!) but this makes the guide more "proper" :))

andreyvelich replied:
Sure, I explain abbreviation.

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
-    You can run experiment without specifying the `goal`. In that case, Katib
-    runs experiment until corresponding succeeded trials reaches `maxTrialCount`.
-    `maxTrialCount` parameter is described bellow.
+  You can run experiment without specifying the `goal`. In that case, Katib
```
```suggestion
  You can run the experiment without specifying the `goal`. In that case, Katib
```

"the" or "your"

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
-    runs experiment until corresponding succeeded trials reaches `maxTrialCount`.
-    `maxTrialCount` parameter is described bellow.
+  You can run experiment without specifying the `goal`. In that case, Katib
+  runs experiment until corresponding succeeded trials reaches `maxTrialCount`.
```
```suggestion
  runs the experiment until corresponding succeeded trials reach `maxTrialCount`.
```

"trials" (plural) "reach"

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
 
-* **metricsCollectorSpec**: A specification of how to collect the metrics from
+- **trialTemplate**: The template that defines the trial.
+  You must package your ML training code into a Docker image, as described
```
To avoid repetition ("you must" x 2)
```suggestion
  You need to package your ML training code into a Docker image, as described
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
+
+  You can define the job in YAML format or you can use a
+  [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/).
+  [Here](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/trial-template-configmap.yaml)
+  is an ConfigMap example with trial templates.
```
```suggestion
  For trial templates, check [this example of a ConfigMap](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/trial-template-configmap.yaml)
```

andreyvelich replied:
I will refactor this section in the following PR with doc for: https://github.com/kubeflow/katib/issues/1341, https://github.com/kubeflow/katib/issues/1280.

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
 
-* **nasConfig**: The configuration for a neural architecture search (NAS).
+- **nasConfig**: The configuration for a neural architecture search (NAS).
   Note: NAS is currently in **alpha** with limited support.
```
```suggestion
  **Note**: NAS is currently in **alpha** with limited support.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -41,159 +42,166 @@ the experiment from the UI or from the command line.
-* [Bayesian optimization](#bayesian)
-* [Hyperband](#hyperband)
-* [Tree of Parzen Estimators (TPE)](#tpe-search)
-* [Covariance Matrix Adaptation Evolution Strategy (CMA-ES)](#cmaes)
-* [Neural Architecture Search based on ENAS](#enas)
-* [Differentiable Architecture Search (DARTS)](#darts)
+- [Grid search](#grid-search)
+- [Random search](#random-search)
+- [Bayesian optimization](#bayesian)
+- [Hyperband](#hyperband)
+- [Tree of Parzen Estimators (TPE)](#tpe-search)
+- [Covariance Matrix Adaptation Evolution Strategy (CMA-ES)](#cmaes)
+- [Neural Architecture Search based on ENAS](#enas)
+- [Differentiable Architecture Search (DARTS)](#darts)
```
Suggestion - we should add this list of available NAS algos to the beginning of the Katib ABOUT page somewhere. Then add "more algorithms are under development."

andreyvelich replied:
Sure, I have added: https://github.com/kubeflow/website/blame/b946be02fb367f8e9c2fb5d88a7818ddd7c18ee8/content/en/docs/components/katib/overview.md#L20-L24. WDYT @8bitmp3 ?

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -666,103 +683,104 @@ You can run a Katib experiment from the command line or from the Katib UI.
 
-* See how to run the random algorithm and other [Katib
-  examples](/docs/components/hyperparameter-tuning/hyperparameter/#random-algorithm)
+- See how to run the random algorithm and other [Katib
+  examples](/docs/components/katib/hyperparameter/#random-algorithm)
   in the getting-started guide.
```
```suggestion
- Learn how to run the [random algorithm and other Katib
  examples](/docs/components/katib/hyperparameter/#random-algorithm).
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -666,103 +683,104 @@ You can run a Katib experiment from the command line or from the Katib UI.
 
-* For a detailed instruction of the Katib Configuration file, 
-  read the [Katib config page](/docs/components/hyperparameter-tuning/katib-config/).
+- For a detailed instruction of the Katib Configuration file,
+  read the [Katib config page](/docs/components/katib/katib-config/).
```
```suggestion
- Read about [Katib Configuration (Katib config)](/docs/components/katib/katib-config/).
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -666,103 +683,104 @@ You can run a Katib experiment from the command line or from the Katib UI.
+  read the [Katib config page](/docs/components/katib/katib-config/).
 
-* See how you can change installation of Katib component in the [environment variables guide](/docs/components/hyperparameter-tuning/env-variables/).
+- See how you can change installation of Katib component in the [environment variables guide](/docs/components/katib/env-variables/).
```
```suggestion
- [How to set up environment variables](/docs/components/katib/env-variables/) for each Katib component.
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -5,165 +5,166 @@ weight = 10
 
-Use Katib for automated tuning of your machine learning (ML) model's 
-hyperparameters and architecture.
+Katib is a Kubernetes-native project for automated machine learning (AutoML).
+For example, to run hyperparameter tuning or neural architecture search.
+Learn more about AutoML at [fast.ai](https://www.fast.ai/2018/07/16/auto-ml2/),
+[Google Cloud](https://cloud.google.com/automl),
+[Microsoft Azure](https://docs.microsoft.com/en-us/azure/machine-learning/concept-automated-ml#automl-in-azure-machine-learning) or
+[Amazon SageMaker](https://aws.amazon.com/blogs/aws/amazon-sagemaker-autopilot-fully-managed-automatic-machine-learning/).
+
+Katib is the framework agnostic to machine learning frameworks.
+It can tune hyperparameters of applications written in any language
+of the users' choice and natively supports many machine learning frameworks,
+such as TensorFlow, MXNet, PyTorch, XGBoost, and others.
+
+The [Katib project](https://github.com/kubeflow/katib) is open source.
+The [developer guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md)
+is a good starting point for developers who want to contribute to the project.
```
Can you please use my previous suggestion (old text from the existing Markdown) and throw in the cloud links, so we don't reinvent the wheel? There's quite a lot of review in this PR :)

andreyvelich replied:
@8bitmp3 Sure, I've made section as you said:
https://github.com/kubeflow/website/blame/b946be02fb367f8e9c2fb5d88a7818ddd7c18ee8/content/en/docs/components/katib/overview.md#L11-L31

8bitmp3 commented:
Review still in progress. Good job @andreyvelich ! That's a lot of work 👍 🚀 I've enjoyed learning about Katib

gaocegege reviewed (commented):
/lgtm

Thanks for your contribution! :tada: :+1:

andreyvelich commented:
/hold for the review

andreyvelich commented:
Thanks a lot for the second review @8bitmp3!
I've addressed your comments.

8bitmp3 commented:
Thanks @andreyvelich I appreciate you massive efforts. I'll look at the edits again when I find some time, since there's a lot to review 👍 Meanwhile, here and in your other PRS, can you run the Markdown text through Google Docs and use their grammar and spell checker, just in case? It's a very useful cool tool, since we're all humans here. Cheers!! Ping me (and/or use `/assign @8bitmp3` to get my attention if something is urgent 🎉

andreyvelich commented:
It's great idea @8bitmp3, I've done it and made few changes.
Thank you again for the review!

If you have any other comments, please let me know.
If everything looks good add `/lgtm` please and we merge this PR.
After that, I will rebase and update other PRs.

/assign @8bitmp3

8bitmp3 commented:
Thank you so much @andreyvelich I'll take a look really soon, thanks for the patience.

andreyvelich commented:
Since PR: https://github.com/kubeflow/website/pull/2332 has been merge, @8bitmp3 can you leave your review on the trial template doc in this PR, please? I will make appropriate changes, if it is needed.
I've rebased the PR.
/cc @gaocegege @johnugeorge

andreyvelich commented:
@RFMVasconcelos @8bitmp3 Do you need more time to review this PR or we can move forward and merge it?

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -1,18 +1,24 @@
 ## Katib Controller
 
-This is the table of environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/katib-controller/katib-controller.yaml) deployment.
+These are environment variables for the
```
```suggestion
Below are the environment variables for the
```

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -71,7 +78,9 @@ If you want to use your own DB Manager to report Katib metrics, you can change `
-This is the table of environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1alpha3/ui/deployment.yaml) deployment.
+These are environment variables for the
+[Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml)
+deployment.
```
```suggestion
deployment:
```

8bitmp3 commented on content/en/docs/components/katib/env-variables.md:
```diff
@@ -116,7 +125,9 @@ Katib UI calls Katib DB Manager with the same address expression as Katib Contro
-This is the table of environment variables for the [Katib DB Manager](https://github.com/andreyvelich/katib/blob/doc-katib-config/manifests/v1alpha3/db-manager/deployment.yaml) deployment.
+These are environment variables for the
+[Katib DB Manager](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/db-manager/deployment.yaml)
+deployment.
```
```suggestion
deployment:
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -6,32 +6,33 @@ weight = 30
 +++
 
 This page describes in detail how to configure and run a Katib experiment.
```
```suggestion
This guide describes how to configure and run a Katib experiment.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -88,117 +103,120 @@ These are the fields in the experiment configuration spec:
 
-* **parallelTrialCount**: The maximum number of hyperparameter sets that Katib
-  should train in parallel.
+  Refer to the the
```
```suggestion
  Refer to the
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -88,117 +103,120 @@ These are the fields in the experiment configuration spec:
   each trial, such as the accuracy and loss metrics.
-  See the [details of the metrics collector](#metrics-collector) below.
+  Learn the [details of the metrics collector](#metrics-collector) below.
+  Default metrics collector is `StdOut`.
```
```suggestion
  The default metrics collector is `StdOut`.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -88,117 +103,120 @@ These are the fields in the experiment configuration spec:
+    [`GraphConfig` type](https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/experiments/v1beta1/experiment_types.go#L301-L306).
+
+  - **operations**: The range of operations that you want to tune for your ML model.
+    For each neural network layer the NAS algorithm selects one of the operations
```
Please double check that NAS has been defined in the beginning of the guide somewhere ("neural architecture search (NAS)") 👍

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -88,117 +103,120 @@ These are the fields in the experiment configuration spec:
+  because `maxTrialCount` was reached, you can resume it by increasing
+  `maxTrialCount`. Specify `resumePolicy: LongRunning`, if you want to use this
+  feature. If you don't need to resume the experiment, specify
+  `resumePolicy: Never`. In that case, suggestion resources will be deleted and
```
```suggestion
  `resumePolicy: Never` — in this case, suggestion resources will be deleted and
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -88,117 +103,120 @@ These are the fields in the experiment configuration spec:
 
-Random sampling is an alternative to grid search, useful when the number of 
-discrete variables to optimize is large and the time required for each 
+Random sampling is an alternative to grid search, useful when the number of
```
```suggestion
Random sampling is an alternative to grid search and is used when the number of
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -231,20 +249,21 @@ Katib supports the following algorithm settings:
 The algorithm name in Katib is `bayesianoptimization`.
 
 The [Bayesian optimization](https://arxiv.org/pdf/1012.2599.pdf) method uses
 gaussian process regression to model the search space. This technique calculates
```
```suggestion
Gaussian process regression to model the search space. This technique calculates
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -379,18 +401,19 @@ Katib supports the following algorithm settings:
 {{% alert title="Alpha version" color="warning" %}}
-Neural architecture search is currently in <b>alpha</b> with limited support. 
-The Kubeflow team is interested in any feedback you may have, in particular with 
+Neural architecture search is currently in <b>alpha</b> with limited support.
```
```suggestion
Neural architecture search (NAS) is currently in <b>alpha</b> with limited support.
```

although it must've been predefined but just in case

andreyvelich replied:
Defined here: https://github.com/kubeflow/website/blame/760ec66a6a2327d0b9ae814fa7695dbbb3d537a5/content/en/docs/components/katib/experiment.md#L10.

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -379,18 +401,19 @@ Katib supports the following algorithm settings:
 the [Katib issue tracker](https://github.com/kubeflow/katib/issues).
 {{% /alert %}}
 
 The algorithm name in Katib is `enas`.
```
Maybe move this sentence up so it appears after the section headline (subheading), like in other algorithm descriptions?

```suggestion
The algorithm name in Katib is `enas`.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -474,15 +498,18 @@ Katib supports the following algorithm settings:
-[enas-example-gpu](https://github.com/kubeflow/katib/blob/master/examples/v1alpha3/nas/enas-example-gpu.yaml).
-The example aims to show all the possible operations. Due to the large search 
-space, the example is not likely to generate a good result.
+- As an ENAS example, see the YAML file for the
+  [enas-example-gpu](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/nas/enas-example-gpu.yaml).
+  The example aims to show all the possible operations. Due to the large search
+  space, the example is not likely to generate a good result.
```
```suggestion
- For an ENAS example, check the YAML file — 
  [`enas-example-gpu.yaml`](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/nas/enas-example-gpu.yaml) —
  which attempts to show all possible operations. Due to the large search
  space, the example is not likely to generate a good result.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -594,73 +621,87 @@ Katib supports the following algorithm settings:
-* Information in the Katib repository on [Differentiable Architecture Search](https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1alpha3/nas/darts).
-* As a DARTS example, see the YAML file for the
-[darts-example-gpu](https://github.com/kubeflow/katib/blob/master/examples/v1alpha3/nas/darts-example-gpu.yaml).
+- As a DARTS example, check the YAML file for the
+  [darts-example-gpu](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/nas/darts-example-gpu.yaml).
```
```suggestion
- The DARTS example — 
  [`darts-example-gpu.yaml`](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/nas/darts-example-gpu.yaml).
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -594,73 +621,87 @@ Katib supports the following algorithm settings:
 Your training code can record the metrics into `stdout` or into arbitrary output
-files. Katib collects the metrics using a *sidecar* container. A sidecar is
+files. Katib collects the metrics using a _sidecar_ container. A sidecar is
 a utility container that supports the main container in the Kubernetes Pod.
```
Check if a K8s "pod" is capitalized

andreyvelich replied:
I think so: https://kubernetes.io/docs/concepts/workloads/pods/

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -594,73 +621,87 @@ Katib supports the following algorithm settings:
+     You should specify the path in the `.source.fileSystemPath.path` field.
+     Check the
+     [TFJob example](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/tfjob-example.yaml#L16-L22).
+     Default directory path is `/var/log/katib/tfevent/`.
```
```suggestion
     The default directory path is `/var/log/katib/tfevent/`.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -594,73 +621,87 @@ Katib supports the following algorithm settings:
+     you specify in the `.source.fileSystemPath.path` field. Training container should log metrics to this file.
+     Check the
+     [file metrics collector example](https://github.com/kubeflow/katib/blob/master/examples/v1beta1/file-metricscollector-example.yaml#L15-L22).
+     Default file path is `/var/log/katib/metrics.log`.
```
```suggestion
     The default file path is `/var/log/katib/metrics.log`.
```

RFMVasconcelos commented:
Thank you for the review @8bitmp3 !

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
++++
+title = "Getting started with Katib"
+description = "How to set up Katib and run some hyperparameter tuning examples"
```
The fact that the readers would be learning through examples is probably a given, so I'd suggest:

```suggestion
description = "How to set up Katib and perform hyperparameter tuning"
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+## Katib setup
+
+This section describes some configurations that you may need to add to your
+Kubernetes cluster, depending on the way you're using Kubeflow and Katib.
```
```suggestion
Let's set up and configure Katib on your Kubernetes cluster with Kubeflow.
```

8bitmp3 commented on content/en/docs/components/katib/experiment.md:
```diff
@@ -671,106 +712,110 @@ You can run a Katib experiment from the command line or from the Katib UI.
 
-For example, run the following command to launch an experiment using the
-random algorithm example:
+**Note:** If you deployed Katib as part of Kubeflow, you need to
```
To make it clear here:

```suggestion
**Note:** If you deployed Katib as part of Kubeflow (your Kubeflow deployment should include Katib), you need to
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+                    
++++
+
+This page gets you started with Katib. Follow this guide to perform any
+additional setup you may need, depending on your environment, and to run a few
+examples using the command line and the Katib user interface (UI).
```
"get started" assumes the setup process is a given, I guess, so:

```suggestion
This guide shows how to get started with Katib and run a few
examples using the command line and the Katib user interface (UI)
to perform hyperparameter tuning.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+### Setting up persistent volumes
+
+If you used the [above script](#katib-install) to deploy Katib,
```
```suggestion
If you used the [above-mentioned script](#katib-install) to deploy Katib,
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+### Setting up persistent volumes
+
+If you used the [above script](#katib-install) to deploy Katib,
+you can skip this step. This script deploys PVC and PV on your cluster.
```
Maybe add link to PVC and PV in case someone doesn't know what these are.

```suggestion
you can skip this step. This script deploys [PersistentVolumeClaim (PVC)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) and [PersistentVolume (PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) on your cluster.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+and [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+
+If you're using Katib outside GKE and your cluster doesn't include a
+StorageClass for dynamic volume provisioning, you must create a persistent
+volume (PV) to bind to the persistent volume claim (PVC) required by Katib.
```
I think it's

```suggestion
StorageClass for dynamic volume provisioning, you must create a PersistentVolume (PV) to bind to the PersistentVolumeClaim (PVC) required by Katib.
```

(Maybe)

andreyvelich replied:
I will use PV and PVC, since we mentioned it above.

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
````diff
@@ -0,0 +1,436 @@
+```
+
+The above `kubectl apply` command uses a YAML file
+([`pv.yaml`](https://raw.githubusercontent.com/kubeflow/katib/master/manifests/v1beta1/pv/pv.yaml))
````
```suggestion
— [`pv.yaml`](https://raw.githubusercontent.com/kubeflow/katib/master/manifests/v1beta1/pv/pv.yaml) — 
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+This example randomly generates the following hyperparameters:
+
+- `--lr`: Learning rate. Type: double.
+- `--num-layers`: Number of layers in the neural network. Type: integer.
+- `--optimizer`: Optimizer. Type: categorical.
```
Would it make sense to say "A gradient descent optimizer for neural network training"? So many variables are being optimized during training, especially in AutoML, it may get confusing for a few:

```suggestion
- `--lr`: Learning rate. Type: double.
- `--num-layers`: Number of layers in the neural network. Type: integer.
- `--optimizer`: Gradient descent optimizer for neural network training. Type: categorical.
```

andreyvelich replied:
I think we can re-name it to: "Optimization method to change the neural network attributes".
WDYT ?

8bitmp3 replied:
If Katib (AutoML) is for classic deep learning stuff only right now, then it's trying to update the weights and minimize the loss function using the optimizer (with the `lr` using gradient descent, until there's a better way to optimize).

One of the examples on `kubeflow/katib` says it can be stoch. gradient descent/SGD or Adam, so I think we have to be precise here:

"`--optimizer`: Gradient descent optimizer for neural network training, implements backpropagation. Type: categorical."

Or something along those lines.

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
````diff
@@ -0,0 +1,436 @@
+Check the experiment status:
+
+```shell
+kubectl -n <your user profile namespace> get experiment random-example -o yaml
````
```suggestion
kubectl -n <YOUR_USER_PROFILE_NAMESPACE> get experiment random-example -o yaml
```

(make it stand out a little)

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+- `.currentOptimalTrial.parameterAssignments` is the corresponding hyperparameter set.
+
+As well, `status` shows the experiment's trials with the current statuses.
```
```suggestion
In addition, `status` shows the experiment's trials with their current status.
```

(statuses and status both work, I think(

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+       alt="The Katib menu panel"
+       class="mt-3 mb-3 border border-info rounded">
+
+1. You should see the list of experiments:
```
```suggestion
1. You should be able to view the list of experiments:
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+     class="mt-3 mb-3 border border-info rounded">
+
+1. Click the name of the experiment, **random-example**.
+1. You should see a graph showing the level of validation and train accuracy for various
```
```suggestion
1. There should be a graph showing the level of validation and train accuracy for various
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+### TensorFlow example
+
+If you installed Katib as part of Kubeflow, you can’t run experiments in Kubeflow namespace.
```
```suggestion
If you installed Katib as part of Kubeflow, you can’t run experiments in the Kubeflow namespace.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
````diff
@@ -0,0 +1,436 @@
+   ```
+
+Follow the steps as described for the _random algorithm example_
+[above](#view-ui), to see the results of the experiment in the Katib UI.
````
```suggestion
[above](#view-ui) to obtain the results of the experiment in the Katib UI.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+### PyTorch example
+
+If you installed Katib as part of Kubeflow, you can’t run experiments in Kubeflow namespace.
```
```suggestion
If you installed Katib as part of Kubeflow, you can’t run experiments in the Kubeflow namespace.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
````diff
@@ -0,0 +1,436 @@
+   ```
+
+Follow the steps as described for the _random algorithm example_
+[above](#view-ui), to see the results of the experiment in the Katib UI.
````
```suggestion
[above](#view-ui) to get the results of the experiment in the Katib UI.
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+Follow the steps as described for the _random algorithm example_
+[above](#view-ui), to see the results of the experiment in the Katib UI.
+
+## Cleanup
```
```suggestion
## Cleaning up
```

8bitmp3 commented on content/en/docs/components/katib/hyperparameter.md:
```diff
@@ -0,0 +1,436 @@
+
+## Cleanup
+
+To delete Katib from Kubernetes cluster run:
```
```suggestion
To remove Katib from your Kubernetes cluster run:
```

8bitmp3 commented on content/en/docs/components/katib/katib-config.md:
```diff
@@ -0,0 +1,218 @@
+                    
++++
+
+This page describes
```
```suggestion
This guide describes
```

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -0,0 +1,205 @@
+[Microsoft Azure](https://docs.microsoft.com/en-us/azure/machine-learning/concept-automated-ml#automl-in-azure-machine-learning) or
+[Amazon SageMaker](https://aws.amazon.com/blogs/aws/amazon-sagemaker-autopilot-fully-managed-automatic-machine-learning/).
+
+Katib supports a lot of various AutoML algorithms, such as
+Bayesian optimization, Tree of Parzen Estimators, Random Search,
+Covariance Matrix Adaptation Evolution Strategy, Hyperband,
+Neural Architecture Search based on ENAS, Differentiable Architecture Search and
+many more. Further algorithms are under development.
```
In the entire paragraph, I'd recommend adding links to arXiv and other original posts that describe each AutoML algorithm in detail (you did that in another guide that listed the algorithms). Also, maybe it's best to spell out ENAS? Just a suggestion.

```suggestion
Katib supports a lot of various AutoML algorithms, such as
Bayesian optimization, Tree of Parzen Estimators, Random Search,
Covariance Matrix Adaptation Evolution Strategy, Hyperband,
Neural Architecture Search based on ENAS, Differentiable Architecture Search and
many more. Additional algorithm support is coming soon.
```

8bitmp3 replied:
Was Google's NAS with RL paper the earliest research published on AutoML, btw? 🤔 "Neural Architecture Search with Reinforcement Learning" - https://arxiv.org/abs/1611.01578

andreyvelich replied:
Yes, it was one of the earliest.
@8bitmp3 Do we want to add it also ?
Our ENAS implementation is based on this paper: https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1beta1/nas/enas#about-the-efficient-neural-architecture-search.

8bitmp3 replied:
It was just a suggestion - thought it could be relevant if you found it some use. Up to you

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -0,0 +1,205 @@
+The [developer guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md)
+is a good starting point for developers who want to contribute to the project.
+
+This page introduces the concepts of hyperparameter tuning, neural
```
```suggestion
This guide introduces the concepts of hyperparameter tuning, neural
```

8bitmp3 replied:
Maybe it makes sense to move this sentence towards the beginning, since every guide starts with: "Dear reader, this doc does XYZ. XYZ is a product that does ABC.."

8bitmp3 commented on content/en/docs/components/katib/overview.md:
```diff
@@ -0,0 +1,205 @@
+is a good starting point for developers who want to contribute to the project.
+
+This page introduces the concepts of hyperparameter tuning, neural
+architecture search, and the Katib system as a component of Kubeflow.
```
It may make sense to define "NAS" early in the guide before referring to it as "NAS"

8bitmp3 reviewed (changes requested):
PTAL

---

In this entire paragraph, I'd recommend adding links to arXiv and other original posts that describe each AutoML algorithm in detail (you did that in another guide that listed the algorithms). Also, maybe it's best to spell out ENAS? Just a suggestion.

Was Google's NAS with RL paper the earliest research published on AutoML, btw? 🤔 "Neural Architecture Search with Reinforcement Learning" - https://arxiv.org/abs/1611.01578

```suggestion
Katib supports a lot of various AutoML algorithms, such as
Bayesian optimization, Tree of Parzen Estimators, Random Search,
Covariance Matrix Adaptation Evolution Strategy, Hyperband,
Neural Architecture Search based on ENAS, Differentiable Architecture Search and
many more. Additional algorithm support is coming soon.
```
---

Would it make sense to say "A gradient descent optimizer for neural network training"? So many variables are being optimized during training, especially in AutoML, it may get confusing for a few:

```suggestion
- `--lr`: Learning rate. Type: double.
- `--num-layers`: Number of layers in the neural network. Type: integer.
- `--optimizer`: Gradient descent optimizer for neural network training. Type: categorical.
```
---

Btw, maybe add links to PVC and PV terms in case someone doesn't know what these are:

```suggestion
you can skip this step. This script deploys [PersistentVolumeClaim (PVC)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) and [PersistentVolume (PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) on your cluster.
```
---

And "Gaussian" may have a capital G 😃 Maybe I'm wrong.

Cheers

8bitmp3 commented:
@andreyvelich Thanks for the hard work. Please check all pages on Netlify (https://deploy-preview-2312--competent-brattain-de2d6d.netlify.app/ - I think)

andreyvelich commented:
Thank you for the another review @8bitmp3 !
I've addressed your comments.
Preview: https://deploy-preview-2312--competent-brattain-de2d6d.netlify.app/.

8bitmp3 commented:
@andreyvelich Thanks. If you're OK with the suggestions/own changes, feel free to get the LGTM and approve labels from the AutoML/Katib experts (which I am not 😄)

/assign @animeshsingh @Bobgy @joeliedtke
(approvers)

andreyvelich commented:
@8bitmp3 Thank you again for the help on this PR!
If you don't have any other suggestions on the [`trial-template`](https://github.com/kubeflow/website/blob/f1d4c0a2ea27f0cfdb291e65a1afd5effa868e6b/content/en/docs/components/katib/trial-template.md) doc, I'll remove `/hold` label and we can merge this PR.

/cc @gaocegege @johnugeorge

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -1,13 +1,13 @@
 +++
 title = "Trial Template Overview"
```
```suggestion
title = "Overview of trial template parameters in Katib"
```

(Sentence case, unless it's a product name, I think)

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -1,13 +1,13 @@
 +++
 title = "Trial Template Overview"
 description = "How to specify trial template parameters and support custom CRD in Katib"
```
```suggestion
description = "How to specify trial template parameters and support a custom resource (CRD) in Katib"
```

(I had to look up CRD on the K8s docs - correct me if I'm wrong. Just wanted to help some new users)

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -1,13 +1,13 @@
-in Katib. Follow this guide to know more about changing trial template specification,
+This guide describes how to configure trial template parameters and use custom
+[Kubernetes CRD](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+in Katib. Follow this page to know more about changing trial template specification,
 how to use [Kubernetes ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/)
 to store templates and how to modify Katib controller to support your
 Kubernetes CRD in Katib experiments.
```
```suggestion
in Katib. You will learn about changing trial template specification,
how to use [Kubernetes ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/)
to store templates and how to modify Katib controller to support your
Kubernetes CRD in Katib experiments.
```

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -232,12 +235,12 @@ to know more about it.
 (e.g. [Kubernetes `Deployment`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/))
 as a trial worker without modifying Katib controller source code and building
-the new image. As long as your CRD creates Kubernetes pods, allows to inject
+the new image. As long as your CRD creates Kubernetes Pods, allows to inject
```
is it Pods or pods in K8s? I thought it's "pods" 😄

andreyvelich replied:
I think it's called `Kubernetes Pods` :). Ref: https://kubernetes.io/docs/concepts/workloads/pods/#using-pods.

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -232,12 +235,12 @@ to know more about it.
+the new image. As long as your CRD creates Kubernetes Pods, allows to inject
 the [sidecar container](https://kubernetes.io/docs/concepts/workloads/pods/) on
-these pods and has succeeded and failed status, you can use it in Katib.
+these Pods and has succeeded and failed status, you can use it in Katib.
```
```suggestion
these Pods and has succeeded and failed status, you can use it in Katib.
```

is it Pods or pods in K8s? I thought it's "pods" 😄

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -247,11 +250,11 @@ Follow these two simple steps to integrate your custom CRD in Katib:
 1. Modify Katib controller
    [ClusterRole's rules](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/rbac.yaml#L5)
-   with the new rule to get access for all resources that are created
+   with the new rule to give Katib access for the all resources that are created
```
Correction:

```suggestion
   with the new rule to give Katib access to all resources that are created
```

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
```diff
@@ -247,11 +250,11 @@ Follow these two simple steps to integrate your custom CRD in Katib:
    [ClusterRole's rules](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/rbac.yaml#L5)
-   with the new rule to get access for all resources that are created
+   with the new rule to give Katib access for the all resources that are created
    by trial. To know more about ClusterRole, check
```
"by the trial"?

8bitmp3 commented on content/en/docs/components/katib/trial-template.md:
````diff
@@ -293,7 +296,7 @@ Expected output for the Tekton `Pipeline`:
 ```
 
 If the above steps are successful, you are able to use your custom object YAML
````
```suggestion
If you ran the above steps successfully, you should be able to use your custom object YAML
```

8bitmp3 reviewed (changes requested):
@andreyvelich Just took a last look at `katib/trial-template.md`. We can revise more stuff in another PR, if necessary. Let's try to get a LGTM and `/approve` soon. Thanks for the patience

andreyvelich commented:
> @andreyvelich Just took a last look at `katib/trial-template.md`. We can revise more stuff in another PR, if necessary. Let's try to get a LGTM and `/approve` soon. Thanks for the patience

Sure @8bitmp3.
/hold cancel
/approve

@gaocegege @johnugeorge Can you give your `/lgtm` please?

andreyvelich commented:
I think we need to get `/approve` from one of the root OWNERS for this PR.
@animeshsingh @Bobgy @joeliedtke Can you help with it, please ?

Bobgy commented:
@andreyvelich 
Can stakeholders lgtm first? I can help with the approval?

Bobgy commented:
/approve

RFMVasconcelos commented:
/lgtm

RFMVasconcelos commented:
Thank you @andreyvelich for this big push!
